### PR TITLE
Orange tree project. requires colorize gem. breakfast + lunch

### PR DESCRIPTION
**Pull request by ElsaTKO**
I completed the orange tree project through the lunch tier. This project requires the colorize gem. I didn't want to hard code the age/orange figures, but my group had more important things to worry about at the time (like figuring out how to make the grove) and we never got back around to making it a nice formula, such as exponential orange growth based on age or something.  So instead it's a very short-lived tree with a very arbitrary yield. Ahh, testing.  :)

### CODE REVIEW

File: orange_grove_final_colorize.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  # user passes number of trees in grove
+  def plant_grove(num_trees)
+    num_trees.times do
+      @grove.push(OrangeTree.new)
+    end
+  end
+
+  # counts all oranges on all trees in array
+  def count_all_the_oranges
```
Comment (davidmfoley) [inline orange_grove_final_colorize.rb:20]: Thought experiment:

I think there is a sequence of events where `grove.total_oranges` and `grove.count_all_the_oranges` could return different values. 

When might that happen and why?

File: orange_grove_final_colorize.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    return @orange_count
+  end
+
+  # reduces orange count by one
+  def pick_an_orange
+    # if loop to check if any oranges left
+    # if there are still oranges, then subtract from orange count and print yum statement
+    if @orange_count > 0
+      @orange_count -= 1
+      puts "You picked an " + "orange".colorize(:yellow) + "! There are now " + "#{@orange_count} oranges".colorize(:yellow) + " left!"
```
Comment (davidmfoley) [inline orange_grove_final_colorize.rb:86]: Per the assignment, these should be returned, not puts.

File: orange_grove_final_colorize.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+      if @age <= 2
+        @orange_count = 0
+      elsif @age <= 5
+        @orange_count = (@age * 10)
+      else
+        @is_dead = true
+        puts "The tree is dead.".colorize(:red) + ":("
+        @orange_count = 0
+        exit
```
Comment (davidmfoley) [inline orange_grove_final_colorize.rb:69]: I don't think `exit` is what you want here - what happens if one tree in a grove dies?
http://ruby-doc.org/core-2.2.2/Kernel.html#method-i-exit

**Comment by davidmfoley** [comment]
colorize == fancy!

File: orange_grove_final_colorize.rb (hunk 3 above)
Comment (ElsaTKO) [inline orange_grove_final_colorize.rb:69]: True. We coded under the assumption the grove would only be planted once with a set of same-age trees.

File: orange_grove_final_colorize.rb (hunk 2 above)
Comment (ElsaTKO) [inline orange_grove_final_colorize.rb:86]: At that time we were using them interchangeably in class, but yeah, it should be returned.

File: orange_grove_final_colorize.rb (hunk 1 above)
Comment (ElsaTKO) [inline orange_grove_final_colorize.rb:20]: I guess I'm not seeing it. We reset and recount @total_oranges for both one_year_passes_grove and count_all_the_oranges methods.

**Pull request merged**